### ISSUE TITLE
Multi-user support with different libaries

### ISSUE THREAD

**Issue by b3nd1k** [monicahq/monica#5988]
Hey! I'd love to see multi-user support where each user get their own library with their own set of contacts, notes, reminders and so on. That way, I would not have to spin up multiple instances if I wanted my family to take advantage of Monica as well, without having access to my data.

**Comment by djaiss** [comment]
This is exactly what we will do in the next version of Monica. We'll call this the notion of vaults. More details coming soon 😄

**Comment by codemonkeysoftware** [comment]
Is this different from a user registering for an account on the web UI or creating an account with `artisan account:create`?

**Comment by djaiss** [comment]
Yeah it's completely different. Right now in Monica we can't have multiple users in the same account with **different** set of contacts. They have to share the exact same data - which might be a problem.

**Comment by codemonkeysoftware** [comment]
> That way, I would not have to spin up multiple instances if I wanted my family to take advantage of Monica as well

@b3nd1k are you self-hosting Monica or using the web service? This makes it sound like you're looking to have multiple users on a self-hosted instance, am I mistaken?

If you're on a self-hosted instance, you can create additional user accounts either by opening registration or using the `artisan` command mentioned above. If you are speaking about monicahq.com, it might be a better solution to just create additional users since you don't want to share your data with them anyway.

**Comment by wunter8** [comment]
Any updates or timeline on this? I'd like to self-host Monica and would prefer not using a separate domain and instance for each separate user
